pkg/scheduler/scheduler.go

```diff
@@ -324,23 +325,46 @@ type partialAssignment struct {
 	preemptionTargets []*workload.Info
 }
 
+// resourcesToReserve calculates how much of available resources in cq assignment could reserve.
+func resourcesToReserve(e *entry, cq *cache.ClusterQueue) cache.FlavorResourceQuantities {
+	if e.CanBePartiallyAdmitted() || e.assignment.Borrowing {
+		return e.assignment.Usage
+	}
+	reservedUsage := e.assignment.Usage
+	for flavor, resourceUsage := range e.assignment.Usage {
+		cqQuota := make(map[corev1.ResourceName]*cache.ResourceQuota)
+		for _, rg := range cq.ResourceGroups {
+			for _, cqFlavor := range rg.Flavors {
+				if cqFlavor.Name == flavor {
+					cqQuota = cqFlavor.Resources
+				}
```

Comment:
Once the scheduler hits here, can we break to Line 343?
It seems that the scheduler doesn't need to continue loops.